docs/bmo-part-of-capm3.md

```diff
@@ -0,0 +1,182 @@
+│   └── kustomization.yaml
+├── ironic_ci.env
+├── namespace
+│   ├── kustomization.yaml
+│   └── namespace.yaml
+├── operator
+│   ├── bmo.yaml
+│   └── kustomization.yaml
+├── rbac
+│   ├── kustomization.yaml
+│   ├── role_binding.yaml
+│   ├── role.yaml
+│   └── service_account.yaml
+└── role.yaml -> rbac/role.yaml
+
+```
+
+As we can see, the deploy directory has one top level folder for deployment,
+namely `default` and it deploys only baremetal operator through kustomization.
+In addition, `crds`, `namespace` and `rbac` folders have their own kuztomization
+and yaml files. The following directory tree visualize a new structure of ironic
+-deployment folder in more detail:
+
+## Proposed structure of ironic-deployment folder
+
+```diff
+tree ironic-deployment/
+
+ironic-deployment/
+├── default
+│   ├── ironic_bmo_configmap.env
+│   └── kustomization.yaml
+├── ironic
+│   ├── ironic.yaml
+│   └── kustomization.yaml
+└── keepalived
+    ├── ironic_bmo_configmap.env
+    ├── keepalived_patch.yaml
+    └── kustomization.yaml
+
+```
+
+Whereas, ironic-deployment folder has three top level folder for deployments,
+namely `default`,  `ironic` and `keepalived`. `default` and `ironic` deploy
+only ironic, `keepalived` deploys the ironic with keepalived. As the name
+implies, `keepalived/keepalived_patch.yaml` patches the default image URL
+through kustomization.
+
+To this end, "production", "local developer" and "ironic outside of the
+cluster" use cases will be met, as new structure will fulfill the needs of those
+cases with cleaner and leaner organization of deployments in place.
+
+### Pivoting use case
+
+Once above outlined folder structural changes are made, we can deploy BMO on its
+own in CAPM3 and Ironic independently to meet the use case of pivoting, where
+BMO should deployed as part of CAPM3. The main concern here is that, BMO and
+Ironic are separated and as a consequence, it will result in breaking the chain
+of workflow, where, previously, BMO could use localhost to talk to ironic
+without any authentication. However, considering that Ironic runs in host
+network, BMO would be able to interact with it, from anywhere that is connected
+or routed to the provisioning network. Therefore, separating Ironic in its own
+pod should not impact BMO in this case.
+
+Furthermore, to support above provided statement, CAPM3  needs to embed BMO in a
+way that it is deployed and modified properly by `clusterctl` without having to
+embed ironic as part of the clusterctl. Thus, we created
+`cluster-api-provider-metal3/config/bmo/` folder to be able to deploy BMO in
+CAPM3. The following directory tree visualizes the  structure in more detail:
+
+```diff
+tree config/bmo/
+
+config/bmo/
+├── bmo_image_patch.yaml
+├── kustomization.yaml
+└── README.md
+
+```
+
+As we can see, the `config/bmo/` deploys `rbac`, `crds` and `operator` using
+cross repository referencing link and `bmo_image_patch.yaml` patches the default
+image URL through kustomization.
```

Comment:
maybe giving the kustomization file would be clearer than this as we can not see that it deploys rbac, crds and operator

Reply:
Agree with @maelk's comment. On top, I would also like to know a bit more about this cross repository referencing way as a reader since I am still confused by reading this that how the changes in api in the baremetal-operator, will be reflected in both locations (repositories)

Reply:
Updated